### PR TITLE
Sprint 48: TLT-1767: Class roster tool

### PR DESCRIPTION
**Pull request by elliottyates**
The Class Roster app with accompanying url routes, views, and settings.

This code needs the following updates to icommons_rest_api to work:
https://github.com/Harvard-University-iCommons/icommons_rest_api/pull/76

This PR does not include unit tests, which will come in a follow-up PR.

### CODE REVIEW

File: class_roster/views.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+def _get_roster_url(course_instance, user_id):
+    calendar_year = str(
+        course_instance.get('term', {}).get('calendar_year', '')).strip()
+    # cs_class_number can be None or blank
+    cs_class_number = course_instance.get('cs_class_number', '')
+    cs_class_number = str(cs_class_number).strip() if cs_class_number is not None else ''
+    term_code = str(
+        course_instance.get('term', {}).get('term_code', '')).strip()
+    if calendar_year == '' or cs_class_number == '' or term_code == '':
+        return None  # we cannot build a link for this course instance; skip it
```
Comment (sapnamysore) [inline class_roster/views.py:72]: Could  we log this?

File: class_roster/views.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        SIS_ROSTER['base_path'],
+        SIS_ROSTER['static_path'],
+        SIS_ROSTER['base_query'],
+        dynamic_query)
+
+    return site_url
+
+
+def _get_course_instances(primary_course_instance_id):
+    # returns primary course instance and all secondary xlisted course instances
```
Comment (sapnamysore) [inline class_roster/views.py]: Can this be in a doctring?

**Comment by sapnamysore** [comment]
Looks good 👍 , just a couple of minor things. Nice work!

Please share the sis_roster_base_path and sis_roster_base_url offline so we can setup the tool locally.

File: class_roster/views.py (hunk 2 above)
Comment (elliottyates) [inline class_roster/views.py]: Sure, I didn't realize there was a difference?

File: class_roster/views.py (hunk 1 above)
Comment (elliottyates) [inline class_roster/views.py:72]: Good idea.

**Comment by elliottyates** [comment]
secure.py details shared on #scrumdev and in the ticket: https://jira.huit.harvard.edu/browse/TLT-1767

**Comment by eparker71** [comment]
👍

**Pull request merged**